### ISSUE TITLE
Inserting an inline object could preserve selection attributes

### ISSUE THREAD

**Issue by oleq** [ckeditor/ckeditor5#15514]
## 📝 Provide a description of the improvement

You can reproduce it with any inline object but I'll be using MathType as an example

0. Visit https://ckeditor.com/docs/ckeditor5/latest/features/math-equations.html#demo.
1. Create a paragraph of text.
2. Bold it (or apply inline styles such as font size, color, background color).
3. Without changing the selection, insert an **inline** object (math equation).
4. Type without changing the selection.

## Proposed

Typing continues previously used selection (content) attributes (bold, font, color, etc.). For instance MS Word and Apple Pages act this way.

## Actual

The user is typing using raw text.

https://github.com/ckeditor/ckeditor5/assets/1099479/4c94db0b-5e29-4d16-9ecf-5d1dfd787604

## Changes to the code

Somewhere around here https://github.com/ckeditor/ckeditor5/blob/c2a54f980f397363048f56385c4ab169fdbbef68/packages/ckeditor5-engine/src/model/utils/insertobject.ts#L112-L125.

## Extras

In the past, we advised integrators how to preserve formatting while inserting an inline object https://github.com/wiris/html-integrations/issues/811#issuecomment-1752677948. This is a slightly different case (inserting into a formatted paragraph, not at the end of it) yet since we're around this topic I'm wondering if this could be automated on the engine's side too. This would spare integrators some unnecessary code. I'm not sure why 

https://github.com/ckeditor/ckeditor5/blob/c2a54f980f397363048f56385c4ab169fdbbef68/packages/ckeditor5-engine/src/model/utils/insertobject.ts#L113

is not enough, though.

So, to sum it up, there would be two cases to handle:

* a selection after insertion should retain attributes from before inline object insertion (this issue),
* an object attempts to inherit all selection attributes it can upon being inserted (a generic approach for the case from Wiris issue tracker)

## Drawbacks

?

---

If you'd like to see this improvement implemented, add a 👍 reaction to this post.
